lib/main.dart

```diff
@@ -17,14 +18,25 @@ void main() async {
     options: DefaultFirebaseOptions.currentPlatform,
   );
 
-  runApp(ProviderScope(
-      child: MaterialApp(
-    title: 'Data Generator',
-    themeMode: ThemeMode.dark,
-    theme: lightTheme,
-    darkTheme: darkTheme,
-    home: TheApp(),
-  )));
+  runApp(ProviderScope(child: MainApp()));
+}
+
+class MainApp extends ConsumerWidget {
+  const MainApp({
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    bool isDarkTheme = ref.watch(themeStateNotifierProvider);
```

Comment:
Not sure why you decide to create new notifier class every time, that's what I created generic one for. So we wouldn't have to create a new class every time we need to save something in the state.